### PR TITLE
Added DataGrid a11y documentation

### PR DESCRIPTION
**Pull request by stitot**

### CODE REVIEW

File: docs/datagrid/accessibility.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    }
+  }
+}
+```
+
+Setting `accessibility.enabled: false` disables all a11y features, including ARIA attributes and any [ARIA live announcements](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions).
+
+All `accessibility` options are optional, and options in `announcements` override the global `enabled` setting.
+
+## Localization
```
Comment (karolkolodziej) [inline docs/datagrid/accessibility.md:27]: Maybe?

```suggestion
## Globalization
```

File: docs/datagrid/accessibility.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        announcements: {
+          ascending: "Sorted ascending",
+          // Additional announcements can be added here
+        }
+      }
+    }
+  }
+}
+```
+
```
Comment (karolkolodziej) [inline docs/datagrid/accessibility.md:52]: Maybe it is worth mentioning how the `setOptions` should be used for that.

File: docs/datagrid/accessibility.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+For a complete list of available options, check out the [API reference](?).
+
+## Header descriptions
+
+You can add an `aria-description` attribute to individual `<th>` table headers by configuring `header[].accessibility.description`. This is especially useful when [grouped headers](https://www.highcharts.com/docs/datagrid/header) are used, as descriptions of header groups provide additional clarity for users relying on assistive technologies.
+
+## Caption vs. headers
+
+Screen readers often skip table captions (`<caption>`) when users scan a page's headings for structure. To ensure accessibility, pair a table caption with a proper heading (`<h1>`–`<h6>`) placed above the table. This approach ensures that users relying on screen readers can understand the table's context while navigating the page.
+
```
Comment (karolkolodziej) [inline docs/datagrid/accessibility.md:62]: IMO, this article should end with a fully featured demo with examples of how to configure it.
I'm not sure if this should be part of this task 🤷

**Review by karolkolodziej — APPROVED** [review]

**Review by ddragula — APPROVED** [review]
Thanks! I will add all the API changes according to this article.

**Pull request merged**